fix: update crystal version to avoid dependency issue

drujensen opened the pull request:
Fixes failed dependency issue:
```
14:57 $ shards update
Resolving dependencies
Fetching https://github.com/amberframework/amber-router.git
Fetching https://github.com/crystal-lang/json_mapping.cr.git
Fetching https://github.com/crystal-ameba/ameba.git
Unable to satisfy the following requirements:

- `crystal (>= 0.35.0)` required by `amber_router 0.4.4`
- `crystal (~> 0.34, >= 0.34.0)` required by `json_mapping 0.1.0`
- `crystal (>= 0.35.0)` required by `ameba 0.14.2`
Failed to resolve dependencies, try updating incompatible shards or use --ignore-crystal-version as a workaround if no update is available.
```

drujensen committed:
fix: update crystal version to avoid dependency issue

Blacksmoke16 commented on shard.yml:
```diff
@@ -4,6 +4,6 @@ version: 0.1.0
   - Brian J. Cardiff <bcardiff@manas.tech>
 
-crystal: 0.34.0
+crystal: 1.0.0
```
```suggestion
crystal: '>= 0.34.0'
```

Otherwise this PR removes compatibility with `0.34.x`, `0.35.x`, and `0.36.x`

drujensen committed:
Update shard.yml

Co-authored-by: George Dietrich <george@dietrich.app>

drujensen committed:
revert back to crystal 1.0.0

I tested the proposed solution of using >= 0.34.0 for backward
compatibility but it doesn't work.  Shards update still complains that
its not compatible which seems like a bug in the shard utility.

drujensen commented:
@bcardiff @Blacksmoke16 I tried locally both `>= 0.34.0` and `1.0.0` and it still complains in both cases.

How do we fix this?
```
Unable to satisfy the following requirements:
- `crystal (~> 1.0, >= 1.0.0)` required by `json_mapping 0.1.0`
```

```
19:07 $ crystal --version
Crystal 1.0.0 (2021-03-22)

LLVM: 9.0.1
Default target: x86_64-apple-macosx
```

gabojkz commented:
do you know why the test failure?

bcardiff commented:
Resolved by #5
